sources/web/datalab/polymer/modules/template-manager/template-manager.ts

```diff
@@ -36,44 +34,47 @@ class NotebookTemplate {
   }
 
   /**
-   * Escapes all regex modifier characters in the given string.
+   * Inserts a new cell at position zero that defines python variables
+   * for all of the parameters.
    */
-  private static _regexEscapeAll(s: string) {
-    return s.replace(/([.*+?^${}()|\[\]\/\\])/g, '\\$1');
-  }
-
-  /**
-   * Substitutes all placeholders in the given notebook's cells with their
-   * values.
-   */
-  public populatePlaceholders(notebook: NotebookContent) {
-
-    notebook.cells.forEach((cell: NotebookCell) => {
-      this.parameters.forEach((parameter: TemplateParameter) => {
-        const placeholder = PLACEHOLDER_PREFIX + parameter.placeholder;
-        const escapedPlaceholder = NotebookTemplate._regexEscapeAll(placeholder);
-        const regex = new RegExp(escapedPlaceholder);
-        const value = '\'' + parameter.value.toString() + '\'';
-        cell.source = cell.source.replace(regex, value);
-      });
-    });
-
+  public addParameterCell(notebook: NotebookContent) {
+    const definitionLines = this.parameters.map(parameter =>
+        parameter.placeholder + ' = ' + JSON.stringify(parameter.value));
+    const header = '# Parameter definitions generated by code';
```

Comment:
Can we make the message a little clearer, maybe "Auto-generated parameter definitions"?

Reply:
Also, do we want this to be a header? I would say this cell shouldn't stand out too much, so as not to interfere with the notebook's flow for the reader.

Reply:
Message updated.
I thought having a comment indicating generated code was a good thing to have. What would you suggest?

Reply:
Oh, I mistook this for a separate markdown cell, so I thought this would be a level-0 header. Never mind my comment.